doc/source/cookbook.rst

```diff
@@ -112,6 +112,27 @@ directory in ``BLD_DIR``, rather than next to the sources. Since
     :start-after: #BEGIN cmake_gcovr
     :end-before: #END cmake_gcovr
 
+.. _support keil uvision format:
+
+Support of Keil uVision format
+------------------------------
+
+.. _comment of issue 697: https://github.com/gcovr/gcovr/issues/697#issuecomment-1336255837
+.. _Keil uVision documentation: https://developer.arm.com/documentation/101407/0537/Debug-Commands/COVERAGE/COVERAGE-GCOV-Export?lang=en
+
+As mentioned in `comment of issue 697`_
+the format of ``gcov`` file generated by the Keil uVision compiler is not compatible with the
+``gcov`` specification.
+To support coverage data generated by this compiler you have to create the gcov files as documented
+in `Keil uVision documentation`_
+and process them before running ``gcov`` to get the correct format.
+
+.. code-block:: sh
+
+    # Untested because we have no access to Keil uVision compiler
+    find . -name '*.gcov' -exec sed -i '' -e 's/\(\#\#\#\#\#)\#/\1/; /^branch/ { s/executed 0/never executed/; s/executed .*/taken 1/; s/skipped .*/never executed/; }' {} \;
+    gcovr --use-gcov-files .
```

Comment:
Since the transformation is a bit more complex I'd suggest extracting the Sed commands into a separate script. Also, the script should cover all markers for uncovered code. Then:

```rst
Save the following Sed script as ``fix-gcov.sed``:

.. code-block::

   # fix markers for uncovered code:
   # match any of #=%$ repeated 6 times
   s/^\([#=%$]\)\(\1\{5\}\)/\2/

   # fix branch tags
   /^branch/ {
     s/executed 0/never executed/
     s/executed .*/taken 1/
     s/skipped .*/never executed/
   }

Then, apply this Sed script to all gcov files before invoking gcovr:

.. code-block:: sh

   find . -name '*.gcov' -exec sed -i -f fix-gcov.sed {} \;
   gcovr --use-gcov-files

.. warning:: Untested because we have no access to Keil uVision compiler
```